coreneuron/coreneuron.hpp

```diff
@@ -0,0 +1,238 @@
+    std::vector<int> nrn_prop_dparam_size_;
+    std::vector<int> nrn_mech_data_layout_; /* 1 AoS (default), >1 AoSoA, 0 SoA */
+    /* array is parallel to memb_func. All are 0 except 1 for ARTIFICIAL_CELL */
+    std::vector<short> nrn_artcell_qindex_;
+    std::vector<bool> nrn_is_artificial_;
+
+    /**
+     * Net Receive function pointer lookup tables
+     */
+    std::vector<pnt_receive_t> pnt_receive; /* for synaptic events. */
+    std::vector<pnt_receive_t> pnt_receive_init;
+    std::vector<short> pnt_receive_size;
+
+    /**
+     * Holds function pointers for WATCH callback
+     */
+    std::vector<nrn_watch_check_t> nrn_watch_check;
+    /**
+     * values are type numbers of mechanisms which do net_send call
+     * related to NMODL net_event()
+     *
+     */
+    std::vector<int> nrn_has_net_event_;
+    /**
+     * inverse of nrn_has_net_event_ maps the values of nrn_has_net_event_ to the index of
+     * ptntype2presyn
+     */
+    std::vector<int> pnttype2presyn;
+
+
+    std::vector<bbcore_read_t> nrn_bbcore_read_;
+    std::vector<bbcore_write_t> nrn_bbcore_write_;
+
+  public:
+
+    auto& get_memb_funcs() {
+        return memb_funcs;
+    }
+
+    auto& get_memb_func(size_t idx) {
+        return memb_funcs[idx];
+    }
+
+    auto& get_different_mechanism_type() {
+        return different_mechanism_type;
+    }
+
+    auto& get_pnt_map() {
+        return pnt_map;
+    }
+
+    auto& get_ion_write_dependency() {
+        return ion_write_dependency;
+    }
+
+    auto& get_net_buf_receive() {
+        return net_buf_receive;
+    }
+
+    auto& get_net_buf_send_type() {
+        return net_buf_send_type;
+    }
+
+    auto& get_bamech() {
+        return bamech;
+    }
+    auto& get_prop_param_size() {
+        return nrn_prop_param_size_;
+    }
+    auto& get_prop_dparam_size() {
+        return nrn_prop_dparam_size_;
+    }
+    auto& get_mech_data_layout() {
+        return nrn_mech_data_layout_;
+    }
+    auto& get_is_artificial() {
+        return nrn_is_artificial_;
+    }
+    auto& get_artcell_qindex() {
+        return nrn_artcell_qindex_;
+    }
+    auto& get_pnt_receive() {
+        return pnt_receive;
+    }
+    auto& get_pnt_receive_init() {
+        return pnt_receive_init;
+    }
+    auto& get_pnt_receive_size() {
+        return pnt_receive_size;
+    }
+    auto& get_watch_check() {
+        return nrn_watch_check;
+    }
+    auto& get_has_net_event() {
+        return nrn_has_net_event_;
+    }
+    auto& get_pnttype2presyn() {
+        return pnttype2presyn;
+    }
+    auto& get_bbcore_read() {
+        return nrn_bbcore_read_;
+    }
+    auto& get_bbcore_write() {
+        return nrn_bbcore_write_;
+    }
+
+
+    /**
+     * Generate point process IDs for pnt_map starting at 1 (since 0 means no point process)
+     * \return the next available point process ID
+     */
+    int get_next_pointtype() {
+        return pointtype++;
+    }
+
+};
+
+extern CoreNeuron crnrn;
```

Comment:
can we rename this to `cnrn` or `corenrn`?

Reply:
done